src/SVector.jl

```diff
@@ -78,7 +78,12 @@ function static_vector_gen(::Type{SV}, @nospecialize(ex), mod::Module) where {SV
             if length(ex.args) == 2
                 return :($f($SV{$(esc(ex.args[2])), Float64})) # default to Float64 like Base
             elseif length(ex.args) == 3
-                return :($f($SV{$(escall(ex.args[3:-1:2])...)}))
+                if f === :rand && ex.args[3] isa Int && ex.args[3] > 0
+                    # supports calls like rand(Type, n) and rand(sampler, n)), but only if n > 0
+                    return :(_rand(Random.GLOBAL_RNG, $(esc(ex.args[2])), Size($(esc(ex.args[3]))), $SV{$(esc(ex.args[3]))}))
```

Comment:
How about adding the type of the element of the vector?
```suggestion
                if f === :rand && ex.args[3] isa Int && ex.args[3] ≥ 0
                    # supports calls like rand(Type, n) and rand(sampler, n))
                    return :(_rand(Random.GLOBAL_RNG, $(esc(ex.args[2])), Size($(esc(ex.args[3]))), $SV{$(esc(ex.args[3])), Random.gentype($(esc(ex.args[2])))}))
```
This change supports a 0-element vector like this.
```julia
julia> @SVector rand(1:3, 0)
0-element SVector{0, Int64} with indices SOneTo(0)
```

I'm not sure which `Random.gentype` or `eltype` would be preferred. [The documentation](https://docs.julialang.org/en/v1/stdlib/Random/#A-simple-sampler-without-pre-computed-data) recommends adding a method to `eltype`, but the docstring of `eltype` does not say anything about samplers, and `Random.gentype` is not well-documented and may not be stable API. However, perhaps some other packages may define their `gentype` method and the result is different from `eltype`.

I think the purposes of these functions are different; `eltype` for iterators, `gentype` for samplers.
It would be better to make the `gentype` function stable, and encourage people to define their `gentype` method (not `eltype`) to create their sampler.

Reply:
OK, it should work I think.